drivers/javascript/ast.coffee

```diff
@@ -1355,7 +1355,10 @@ rethinkdb.circle = aropt (cen, rad, opts) -> new Circle opts, cen, rad
 
 rethinkdb.uuid = (args...) -> new UUID {}, args...
 
-rethinkdb.range = (args...) -> new Range {}, args...
+rethinkdb.range = (args...) ->
+    if args.length and typeof args[args.length - 1] is 'object'
```

Comment:
On other places we're using this check:
```js
((Object::toString.call(args[args.length - 1]) is '[object Object]') and not (args[args.length - 1] instanceof TermBase))
```

This might make a difference if the `step` argument is a ReQL term itself. Though I haven't tested it.